pkg/util/k8sutil/tpr_util.go

```diff
@@ -110,8 +99,11 @@ func updateClusterTPRObject(k8s *unversioned.Client, host, ns string, e *spec.Et
 		return nil, err
 	}
 	req.Header.Set("Content-Type", "application/json")
+	return sendReq(k8s.Client, req)
+}
 
-	resp, err := k8s.Client.Do(req)
+func sendReq(httpcli *http.Client, req *http.Request) (*spec.EtcdCluster, error) {
```

Comment:
I think our util has somewhat become complicated and will only be more in the future.

I have been researching about using the RESTClient from k8s client. It abstracts and provides specific error. It's also complicated though. But at least we don't need to write that.